sklearn/utils/tests/test_array_api.py

```diff
@@ -371,3 +400,37 @@ def test_get_namespace_array_api_isdtype(wrapper):
 
     with pytest.raises(ValueError, match="Unrecognized data type"):
         assert xp.isdtype(xp.int16, "unknown")
+
+
+@pytest.mark.parametrize(
+    "array_namespace, device, _", yield_namespace_device_int_dtype_combinations()
+)
+@pytest.mark.parametrize("invert", [True, False])
+@pytest.mark.parametrize("assume_unique", [True, False])
+@pytest.mark.parametrize("element_size", [6, 10, 14])
+@pytest.mark.parametrize("int_dtype", ["int32", "int64", "uint8"])
```

Comment:
I don't understand the point of introducing `yield_namespace_device_int_dtype_combinations` if we test only for a fixed, namespace agnostic, subset of dtypes in the end.